OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -220,12 +220,14 @@ void SanityCheckPath(Mobile mobile)
 
 			var nextCell = path[path.Count - 1];
 
+			var containsTemporaryBlocker = WorldUtils.ContainsTemporaryBlocker(self.World, nextCell, self);
+
 			// Next cell in the move is blocked by another actor
-			if (!mobile.CanMoveFreelyInto(nextCell, ignoredActor, true))
+			if (!mobile.CanMoveFreelyInto(nextCell, ignoredActor, true) || containsTemporaryBlocker)
```

Comment:
Would the order of the checks matter here? Like if `containsTemporaryBlockers` is already `true`, we'd save us the `CanMoveFreelyInto` check with this setup:
`if (containsTemporaryBlocker || !mobile.CanMoveFreelyInto(nextCell, ignoredActor, true))`?